### PR TITLE
x/gamm: Remove commented CleanupBalancerPool function

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #1925 

## What is the purpose of the change

Remove commented `CleanupBalancerPool` function in `pool.go`

## Brief Changelog

- Remove commented `CleanupBalancerPool` function in `pool.go`

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not applicable)

### CODE REVIEW

File: x/gamm/keeper/pool.go [hunk 1]
```diff
@@ -104,98 +104,6 @@ func (k Keeper) DeletePool(ctx sdk.Context, poolId uint64) error {
 	return nil
 }
 
-// CleanupBalancerPool destructs a pool and refund all the assets according to
-// the shares held by the accounts. CleanupBalancerPool should not be called during
-// the chain execution time, as it iterates the entire account balances.
-// TODO: once SDK v0.46.0, use https://github.com/cosmos/cosmos-sdk/pull/9611
```
Comment (p0mvn) [inline x/gamm/keeper/pool.go:110]: Seems that there is some context here about bringing this back with SDK v0.46.0

I support removing dead code and reintroducing, once needed. However, we should probably document this in an issue, linking this PR

**Review by p0mvn — APPROVED** [review]
LGTM

File: x/gamm/keeper/pool.go (hunk 1 above)
Comment (ValarDragon) [reply x/gamm/keeper/pool.go:110]: I'd rather keep this personally

**Comment by ValarDragon** [comment]
Perhaps lets uncomment this?

**Comment by AlpinYukseloglu** [comment]
> Perhaps lets uncomment this?
> 
> And flag the todo + testing this in a github issue?

Just to clarify, we want to add removing pools as a feature then, right? So instead of removing `DeletePool` and the commented `CleanupBalancerPool`, we keep `DeletePool` and uncomment `CleanupBalancerPool`?

**Comment by ValarDragon** [comment]
> Just to clarify, we want to add removing pools as a feature then, right? So instead of removing DeletePool and the commented CleanupBalancerPool, we keep DeletePool and uncomment & make tests for CleanupBalancerPool?

Yup!

I think you can edit the issue or post a comment

**Comment by AlpinYukseloglu** [comment]
> > Just to clarify, we want to add removing pools as a feature then, right? So instead of removing DeletePool and the commented CleanupBalancerPool, we keep DeletePool and uncomment & make tests for CleanupBalancerPool?
> 
> Yup!
> 
> I think you can edit the issue or post a comment

Done! Closing this PR since it's no longer relevant

**Pull request closed without merging**